boards/thingy52/include/board.h

```diff
@@ -55,6 +55,21 @@ extern "C" {
 #define LPSXXX_PARAM_ADDR   (0x5c)
 /** @} */
 
+#ifdef MODULE_PWM_DAC
+/**
+ * @name    The emulated DAC channels
+ *
+ * The static value is actually the result of the @ref pwm_dac_init in
+ * board_init, and it is checked there whether the values are identical.
+ *
+ * @{
+ */
+#define DAC_NUMOF 1
+#undef DAC_LINE
+#define DAC_LINE(x) (0x600)
```

Comment:
Now that's awkward to use.

Why not have 

```C
typedef struct {
    pwm_t   pwm;
    uint8_t chan;
    uint8_t res;
} pwm_dac_params_t;
```

and a `static const pwm_dac_params_t pwm_dac_params[]` to which `dac_t` is the index.
Yea, boring just like the other drivers :wink: 

I can't imagine the added overhead to be significant. 

I think then you could even drop `pwm_dac_init()`.

Reply:
A `static const` set up by some compilation unit couldn't be indexed into by the `dac_set` implementation that's supposed to take a `DAC_LINE(0)` to be just like all the other drivers. Here having a driver (usually passed by pointer) and a peripheral (usually passed by index) just clash.

I'm giving it a try by moving the `static const` from pwm_dac_params.h into the C file -- it's not fully in line with how drivers are usually done, but then again it's a bit more like a peripheral implementation than a driver.